pilot/helpers/agents/Developer.py

```diff
@@ -291,17 +358,18 @@ def get_run_command(self, convo):
             self.run_command = single_match.group(1).strip()
 
     def task_postprocessing(self, convo, development_task, continue_development, task_result, last_branch_name):
-        # TODO: why does `run_command` belong to the Developer class, rather than just being passed?
-        #       ...It's set by execute_task() -> task_postprocessing(), but that is called by various sources.
-        #       What is it at step_human_intervention()?
-        self.get_run_command(convo)
-
-        if development_task is not None:
-            convo.remove_last_x_messages(2)
-            detailed_user_review_goal = convo.send_message('development/define_user_review_goal.prompt', {
-                'os': platform.system()
-            }, should_log_message=False)
-            convo.remove_last_x_messages(2)
+        if self.project.last_detailed_user_review_goal is None:
+            self.get_run_command(convo)
+
+            if development_task is not None:
+                convo.remove_last_x_messages(2)
+                detailed_user_review_goal = convo.send_message('development/define_user_review_goal.prompt', {
+                    'os': platform.system()
+                }, should_log_message=False)
+                convo.remove_last_x_messages(2)
+        else:
+            detailed_user_review_goal = self.project.last_detailed_user_review_goal['llm_response']['text'] if\
+                self.project.last_detailed_user_review_goal is not None else None
```

Comment:
I think `self.project.last_detailed_user_review_goal` can never be None since it's checked in the `if` clause above?

Reply:
True, before was checking last_iteration and didn't cleanup properly 👍  updated